src/Nova/Role.php

```diff
@@ -49,6 +49,15 @@ class Role extends Resource
      */
     public static $title = 'name';
 
+    /**
+     * The relationships that should be eager loaded when performing an index query.
+     *
+     * @var array
+     */
+    public static $with = [
```

Comment:
This does not make sense. Every query will load `permissions` relation

Reply:
Right now, an index page is creating 2 additional queries per line. 1 for the users count and another for get the same all permissions.

I added a static cache for "all permissions" in the Role resource class. This way it will make 1 query per index request. 
And added an eager load for the users. 

For 5 Roles (as of now):
<img width="256" alt="image" src="https://github.com/user-attachments/assets/2bd49121-751e-40e6-b6a1-2f36e611d129">

For the same 5, with the optimizations:
<img width="264" alt="image" src="https://github.com/user-attachments/assets/d5be0d4f-0fb9-4868-8d65-f90aee057beb">